### PR TITLE
Fix for https://github.com/doctrine/mongodb-odm/issues/1641

### PR DESCRIPTION
**Pull request by jarkt**
Missing value with mapping type "hash" if key is empty

### CODE REVIEW

**Comment by malarzm** [comment]
I have no authority here but I'm sure @alcaeus will ask you for a test preventing regression in the future ;)

File: lib/Alcaeus/MongoDbAdapter/TypeConverter.php [hunk 1]
```diff
@@ -47,7 +47,7 @@ public static function fromLegacy($value)
             case is_object($value);
                 $result = [];
 
-                foreach ($value as $key => $item) {
+                foreach ((array)$value as $key => $item) {
```
Comment (alcaeus) [inline lib/Alcaeus/MongoDbAdapter/TypeConverter.php]: Space after the type cast please.

**Comment by malarzm** [comment]
@alcaeus also I'm not sure if it's possible at all, but previously `foreach ($value as $key => $item)` could behave differently when given a `\Traversable` object. Unless you're dealing only with `\stdClass`es here only then don't mind this at all :)

**Comment by jarkt** [comment]
It's a little bit difficult because I have no test environment. So I have to setup a new docker container first - or two because of the check against ext-mongo. So it would be good if you can write a test or otherwise you have to wait a while until I have the time.

**Comment by alcaeus** [comment]
I’m on vacation for two weeks, so I’m in no rush. If you manage to write tests until then, fine; if not, I’ll take a look.

**Comment by alcaeus** [comment]
@jarkt Any chance you got around to writing tests?

**Comment by jarkt** [comment]
It needs some minutes to write the test, but hours to setup test environments. I assume you have it already, so it's much easier for you.

**Comment by alcaeus** [comment]
@jarkt while looking at the code to create appropriate test cases I went down the rabbit hole and found a few different issues. Turns out the issue lies somewhere else, so I created #203 to fix this. Thanks for reporting the issue!

**Pull request closed without merging**